services/QuillLMS/app/models/blog_post.rb

```diff
@@ -141,6 +139,10 @@ def add_published_at
     update(published_at: DateTime.current)
   end
 
+  def self.related_posts(blog_post)
```

Comment:
Since the `related_posts` is dependent on the blog_post, you could rewrite this is as an instance method:

```
def related_posts
  BlogPost
    .where(topic: topic).
    .where.not(id: id)
    .order(created_at: :desc)
    .limit(MOST_RECENT_LIMIT)
end
```
This eliminates the parameter passing, but also removes the need to check for nil:
`blog_post&.related_posts`

vs `BlogPost.related_post(blog_post) if blog_post.nil?`

Reply:
yeah, that makes sense-- thanks for the explanation!